flower_classifier/datasets/oxford_flowers.py

```diff
@@ -147,6 +147,29 @@ def download_dataset(root_dir: str):
         torchvision.datasets.utils.download_url(LABELS_URL, root_dir)
 
 
+def split_dataset(root_dir: str, target_dir: str, test_size=0.2):
+    """
+    Creates two CSVs with filepaths to the images in the original dataset.
+    """
+    import pandas as pd
+    from sklearn.model_selection import train_test_split
+
+    root_dir = Path(root_dir)
+    target_dir = Path(target_dir)
+    labels_filename = root_dir / "imagelabels.mat"
+    labels = loadmat(labels_filename)["labels"].flatten() - 1
+    filepaths = [root_dir / "jpg" / f"image_{index+1:05}.jpg" for index in range(len(labels))]
```

Comment:
What if `split_dataset` is called before the data is downloaded? Could we call the `download_dataset` function here to check for the existence of and, if needed, download the data? This would allow `split_dataset` to be used without depending on a preceding `OxfordFlowersDataModule` instance.